tests/x11/evolution/evolution_prepare_servers.pm

```diff
@@ -14,32 +14,36 @@ use strict;
 use base "x11test";
 use testapi;
 use utils;
+use version_utils "is_sle";
 
 sub run() {
-    # check for the SLE version, die if not supported
-    # dovecot repository must be ajusted for versions > 12 to run full
-    # evolution imap/pop tests
-    die 'not supported SLE version' if not get_var('VERSION') =~ /12/;
-
     select_console('root-console');
     pkcon_quit;
 
-    # add SLES repository and install dovecot
-    zypper_call("ar http://download.suse.de/ibs/SUSE:/SLE-12:/Update/standard/ sle_server_repo");
+    my $dovecot_repo = get_required_var("DOVECOT_REPO");
+    # Add dovecot repository and install dovecot
+    zypper_call("ar ${dovecot_repo} dovecot_repo");
+
     zypper_call("--gpg-auto-import-keys ref");
     zypper_call("in dovecot", exitcode => [0, 102, 103]);
+    zypper_call("rr dovecot_repo");
     save_screenshot;
 
     # configure dovecot
     assert_script_run "sed -i -e 's/#mail_location =/mail_location = mbox:~\\/mail:INBOX=\\/var\\/mail\\/%u/g' /etc/dovecot/conf.d/10-mail.conf";
     assert_script_run "sed -i -e 's/#mail_access_groups =/mail_access_groups = mail/g' /etc/dovecot/conf.d/10-mail.conf";
     assert_script_run "sed -i -e 's/#ssl_cert =/ssl_cert =/g' /etc/dovecot/conf.d/10-ssl.conf";
     assert_script_run "sed -i -e 's/#ssl_key =/ssl_key =/g' /etc/dovecot/conf.d/10-ssl.conf";
+    assert_script_run "sed -i -e 's/#ssl_dh =/ssl_dh =/g' /etc/dovecot/conf.d/10-ssl.conf";
     assert_script_run "sed -i -e 's/auth_mechanisms = plain/auth_mechanisms = plain login/g' /etc/dovecot/conf.d/10-auth.conf";
-    assert_script_run "sed -i -e '96,98 s/#//g' /etc/dovecot/conf.d/10-master.conf";
+    my $uncomment_lines = is_sle("15+") ? "107,109" : "96,98";
```

Comment:
looks tricky. no easy way to understand from test code what you actually uncommenting. also, any add/remove lines from 10-master.conf will require an update of this line. Any chance change this to query for some certain text ?

Reply:
I see that you just keep the same behavior as it was before. But still, maybe you can come up with some regex which would do the same without hardcode lines needed to uncomment?

If you insist to keep it like this, could you at least add a comment describing what you actually trying to uncomment?